### PR TITLE
feat: cost explorer integration

### PR DESCRIPTION
**Pull request by pubudug**
Screenshot of how it looks attached

[screenshot-localhost_3000-2021.06.08-10_14_00.pdf](https://github.com/meshcloud/meshcloud-docs/files/6614691/screenshot-localhost_3000-2021.06.08-10_14_00.pdf)

### CODE REVIEW

**Comment by meshcloud-gmbh** [comment]
Task linked: [CU-jk5nvm Introduce AWS Cost Explorer API for AWS Metering](https://app.clickup.com/t/jk5nvm)

File: docs/meshstack.aws.metering.md [hunk 1]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
     }
 }
 ```
+
+## AWS Cost & Usage Report based approach (Deprecated)
+
+meshStack imports metering data from [AWS Cost and Usage Reports](https://aws.amazon.com/aws-cost-management/aws-cost-and-usage-reporting/)
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:160]:
```suggestion
meshStack imports metering data from [AWS Cost and Usage Reports](https://aws.amazon.com/aws-cost-management/aws-cost-and-usage-reporting/).
```

File: docs/meshstack.aws.metering.md [hunk 2]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
...
+
+An AWS Cost and Usage Report should be created as specified in the [documentation](https://docs.aws.amazon.com/cur/latest/userguide/cur-create.html) with the following properties
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:162]:
```suggestion
An AWS Cost and Usage Report should be created as specified in the [documentation](https://docs.aws.amazon.com/cur/latest/userguide/cur-create.html) with the following properties:
```

File: docs/meshstack.aws.metering.md [hunk 3]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
...
+
+* Daily time granularity
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:164]: can we change the terminology here? Is this the interval?

File: docs/meshstack.aws.metering.md [hunk 4]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
...
+* gzip compression
+* txt/csv file format
+* Should include resource ids
+* Versioning should be setup to overwrite existing report
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:168]:
```suggestion
* Versioning should be setup to overwrite existing reports
```

File: docs/meshstack.aws.metering.md [hunk 5]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
...
+
+**Note**: If an S3 bucket is not already set up, it should be created as specified in [AWS documentation](https://docs.aws.amazon.com/cur/latest/userguide/cur-s3.html)
+
+The following parameters are required to configure meshStack to process the AWS Cost and Usage Reports [[More Details](#configuration-reference)]
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:172]:
```suggestion
The following parameters are required to configure meshStack to process the AWS Cost and Usage Reports [[More Details](#configuration-reference)]:
```

File: docs/meshstack.aws.metering.md [hunk 6]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
...
+
+* AWS credentials that can access the S3 bucket where the reports are written
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:174]:
```suggestion
* AWS credentials that can access the S3 bucket where the reports are written to
```

File: docs/meshstack.aws.metering.md [hunk 7]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
...
+* The name of the S3 bucket where the report is written
+* The region of the S3 bucket
+* The name of the report
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:177]: is there a standard name for the report?

File: docs/meshstack.aws.metering.md [hunk 8]
```diff
@@ -253,3 +154,152 @@ If you would like to see the AWS costs attributed to a seller in the [chargeback
...
+* The "Report path prefix" configured when creating the report
+
+When processing the AWS Cost and Usage Report to generate the Usage Report in the meshPanel,
```
Comment (chewbeccy) [inline docs/meshstack.aws.metering.md:180]: this sentence suddenly ends

**Review by chewbeccy — APPROVED** [review]
Some minor comments

File: docs/meshstack.aws.metering.md (hunk 3 above)
Comment (pubudug) [reply docs/meshstack.aws.metering.md:164]: The word granularity is used in AWS docs. I think better keep it https://docs.aws.amazon.com/cur/latest/userguide/cur-create.html

File: docs/meshstack.aws.metering.md (hunk 7 above)
Comment (pubudug) [reply docs/meshstack.aws.metering.md:177]: No. Whoever sets up the report in the AWS console can choose the name.

File: docs/meshstack.aws.metering.md (hunk 8 above)
Comment (pubudug) [reply docs/meshstack.aws.metering.md:180]: It is the start for the two items below. `When processing the AWS Cost and Usage Report to generate the Usage Report in the meshPanel, 1. you can... 2.  Only the line items with`

File: docs/meshstack.aws.metering.md [hunk 9]
```diff
@@ -139,61 +92,18 @@ let AwsPlatformKrakenConfiguration =
...
-              This setting is useful to prevent read timeout errors in the case of accounts with
-              a large number of cost items.
-          -}
-          Natural
-      , apply-discounts :
+      , data-source :
           {-
-              Whether the discounts contained in the AWS Cost and Usage Reports should be applied
-              when generating the tenant usage reports or not
+          Currently both AWS Cost and Uasge Reports and AWS Cost Explorer are supported.
```
Comment (sttomm) [inline docs/meshstack.aws.metering.md:103]:
```suggestion
          Currently both AWS Cost and Usage Reports and AWS Cost Explorer are supported.
```

**Review by sttomm — APPROVED** [review]

**Pull request merged**